### PR TITLE
Added two env vars for enabling query performance

### PR DESCRIPTION
**Pull request by minhtuev**
## What changes are proposed in this pull request?

Added two env vars that control the default behaviors for query performance mode

## How is this patch tested? If it is not, please explain why.

* Manual testing

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
  - Updated terminology in the Options component from "Lightning mode" to "Query Performance mode."
  - Introduced new configuration options related to query performance.

- **Bug Fixes**
  - Improved exception handling in the saved_views method for better reliability.

- **Documentation**
  - Updated GraphQL schema and fragments to include new fields for query performance settings.

- **Chores**
  - Marked the lightning_threshold attribute for deprecation in the configuration settings.

### CODE REVIEW

File: fiftyone/core/config.py [hunk 1]
```diff
@@ -348,12 +348,21 @@ def __init__(self, d=None):
...
+        # deprecate lightning threshold in favor of a timer-based approach
         self.lightning_threshold = self.parse_int(
             d,
             "lightning_threshold",
             env_var="FIFTYONE_APP_LIGHTNING_THRESHOLD",
             default=None,
         )
+        self.enable_query_performance = self.parse_bool(
+            d, "enable_query_performance",
+            env_var="FIFTYONE_APP_ENABLE_QUERY_PERFORMANCE", default=True
```
Comment (kaixi-wang) [inline fiftyone/core/config.py:360]: Why are we defaulting to true for both enable and default?

Also the naming is super confusing. I have no idea what query performance true/false actually does.

Comment (minhtuev) [reply fiftyone/core/config.py:360]: @kaixi-wang : do you have a suggestion for name change?

**Review by benjaminpkane — APPROVED** [review]
LGTM

File: fiftyone/core/config.py (hunk 1 above)
Comment (minhtuev) [reply fiftyone/core/config.py]: FIFTYONE_ENABLE_QUERY_PERFORMANCE=true/false: whether to disable the entire query performance feature, including the panel, toasts, etc; default: true

FIFTYONE_DEFAULT_QUERY_PERFORMANCE=true/false: whether to enable query performance by default for each dataset (if query performance is enabled systemwide per first env var); default: true

**Pull request merged**